pkg/client/client.go

```diff
@@ -54,7 +54,7 @@ const (
 	kubeConfigEnvVar         = "KUBECONFIG"
 )
 
-var newClientMutex sync.Mutex
+var newClientMutex sync.RWMutex
```

Comment:
This would have no effect unless you also use RLock()

Reply:
I rebased my branch on your fix branch. We'll what the test workflow says.